mlflow/store/hdfs_artifact_repo.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+from six.moves import urllib
+
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.validation import path_not_unique, bad_path_message
+
+
+class HdfsArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on HDFS.
+
+    This repository is used with URIs of the form ``hdfs:/<path>``. The repository can only be used
+    together with the RestStore.
+    """
+
+    def __init__(self, artifact_uri):
+        parsed = urllib.parse.urlparse(artifact_uri)
+        self.config = {
+            'host': parsed.hostname,
+            'port': 8020 if parsed.port is None else parsed.port
+        }
+        if "MLFLOW_HADOOP_HOST" in os.environ:
+            self.config['host'] = os.environ['MLFLOW_HADOOP_HOST']
+        if "MLFLOW_HADOOP_PORT" in os.environ:
+            self.config['port'] = os.environ['MLFLOW_HADOOP_PORT']
```

Comment:
Why do we want to load these from environment variables -- shouldn't they be part of the hdfs URI scheme?

Reply:
Thanks for pointing out. The above lines are not required and corrected the same.